src/coq/constants/equtils.ml

```diff
@@ -163,3 +163,44 @@ let is_rewrite_r (trm : types) : bool =
  *)
 let is_rewrite (trm : types) : bool =
   is_rewrite_l trm || is_rewrite_r trm
+
+
+(* Information required to perform a rewrite. *)
+type rewrite_args = {
+    a : types;
+    x : constr;
+    p : constr;
+    px : constr;
+    y : constr;
+    eq : constr;
+    params : constr array;
+    left : bool
+  }
+
+(* Proof of x = x where x : A. *)
+type eq_refl_args = {
```

Comment:
I would use the original eq_refl_app struct here, even with a second function that returns an optional